### PR TITLE
PAYARA-3718 Adds EJB over HTTP client config via system properties

### PR DESCRIPTION
**Pull request by jbee**
### Description
I'm not entirely sure this is what we want but I think this is what we tried to do. I figured it is easiest to do the fallback in the factory. All environment variables of `javax.naming.Context` and our `RemoteEJBContextFactory` are checked for being defined as **system property of the same name**.

### CODE REVIEW

**Comment by jbee** [comment]
jenkins test please

File: appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java [hunk 1]
```diff
@@ -77,9 +77,36 @@
...
+        updateEnvironmentFromSystemProperties((Hashtable<String, Object>) environment);
         return new RemoteEJBContext(environment);
     }
-    
+
+    private void updateEnvironmentFromSystemProperties(Hashtable<String, Object> environment) {
+        for (String key : SYSTEM_PROPERTY_KEYS) {
+            if (!environment.containsKey(key)) {
+                String systemPropertyName = key.replaceFirst("fish\\.payara\\.|java\\.naming\\.", "fish.payara.ejb.http.client.");
+                System.out.println("Checking "+systemPropertyName);
```
Comment (smillidge) [inline appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java]: this should be removed

Comment (jbee) [reply appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java]: :D Forgot to remove it after adding it during testing.

**Comment by jbee** [comment]
@smillidge there is the added array of properties checked. If we want to support all of the `java.naming` ones we can add all of them to the array.

**Comment by jGauravGupta** [comment]
jenkins test please

File: appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java [hunk 2]
```diff
@@ -77,9 +77,35 @@
...
     public Context getInitialContext(Hashtable<?, ?> environment) throws NamingException {
+        updateEnvironmentFromSystemProperties((Hashtable<String, Object>) environment);
         return new RemoteEJBContext(environment);
     }
-    
+
+    private void updateEnvironmentFromSystemProperties(Hashtable<String, Object> environment) {
+        for (String key : SYSTEM_PROPERTY_KEYS) {
+            if (!environment.containsKey(key)) {
+                String systemPropertyName = key.replaceFirst("fish\\.payara\\.|java\\.naming\\.", "fish.payara.ejb.http.client.");
```
Comment (jbee) [inline appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java]: I am starting to think that what users expect is that system properties of the same name as the environment variables are checked. Can anyone confirm?

File: appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java [hunk 3]
```diff
@@ -98,12 +116,11 @@ public Context getInitialContext(Hashtable<?, ?> environment) throws NamingExcep
     }
 
     private void updateEnvironmentFromSystemProperties(Hashtable<String, Object> environment) {
-        for (String key : SYSTEM_PROPERTY_KEYS) {
-            if (!environment.containsKey(key)) {
-                String systemPropertyName = key.replaceFirst("fish\\.payara\\.|java\\.naming\\.", "fish.payara.ejb.http.client.");
-                String value = System.getProperty(systemPropertyName, null);
+        for (String environmentVariable : SYSTEM_PROPERTY_KEYS) {
+            if (!environment.containsKey(environmentVariable)) {
```
Comment (svendiedrichsen) [inline appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java]: The method name `update...` suggests that you can actually update the environment map from system properties. But this if allows only to set missing values.

**Comment by jbee** [comment]
jenkins test please

File: appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java (hunk 3 above)
Comment (jbee) [reply appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java]: renamed it.

**Comment by pdudits** [comment]
jenkins test please

File: appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java [hunk 4]
```diff
@@ -111,9 +111,52 @@
...
     public static final String FISH_PAYARA_CLIENT_ADAPTER = CLIENT_ADAPTER;
 
+    /**
+     * The keys checked when creating a {@link Context} with {@link #getInitialContext(Hashtable)}. If these are not set
+     * in the environment they are initialised to a {@link System#getProperty(String)} if present. 
+     * 
+     * The name of the property is the same as the environment variable name.
+     */
+    private String[] SYSTEM_PROPERTY_KEYS = {
+            Context.INITIAL_CONTEXT_FACTORY,
```
[the remark this answers is not in the document]
Comment (jbee) [reply appserver/ejb/ejb-http-remoting/client/src/main/java/fish/payara/ejb/http/client/RemoteEJBContextFactory.java]: I had the same thought but all of these in the list are defined as some `String` value according to the javadoc that is converted later on.

**Pull request merged**